crates/matrix-sdk-base/src/store/ambiguity_map.rs

```diff
@@ -154,60 +154,76 @@ impl AmbiguityCache {
         self.changes.entry(room_id.to_owned()).or_default().insert(event_id, change);
     }
 
-    async fn get(
-        &mut self,
+    async fn get_old_display_name(
+        &self,
         changes: &StateChanges,
         room_id: &RoomId,
-        member_event: &SyncRoomMemberEvent,
-    ) -> Result<(Option<DisplayNameUsers>, Option<DisplayNameUsers>)> {
+        new_event: &SyncRoomMemberEvent,
+    ) -> Result<Option<String>> {
         use MembershipState::*;
 
         let old_event = if let Some(m) = changes.state.get(room_id).and_then(|events| {
-            events.get(&StateEventType::RoomMember)?.get(member_event.state_key().as_str())
+            events.get(&StateEventType::RoomMember)?.get(new_event.state_key().as_str())
```

Comment:
Could we store `new_event.state_key()` into a local variable named `user_id`, and reuse it all over the place in this function? I think it would make reading this code much simpler.

Reply:
Yup 8c5fdbbbe479825e081afd67e5db9f82d50c33a8.